lib/passport-bitbucket/strategy.js

```diff
@@ -42,55 +47,75 @@ var util = require('util')
  */
 function Strategy(options, verify) {
   options = options || {};
-  options.requestTokenURL = options.requestTokenURL || 'https://bitbucket.org/api/1.0/oauth/request_token/';
-  options.accessTokenURL = options.accessTokenURL || 'https://bitbucket.org/api/1.0/oauth/access_token/';
-  options.userAuthorizationURL = options.userAuthorizationURL || 'https://bitbucket.org/api/1.0/oauth/authenticate/';
-  options.sessionKey = options.sessionKey || 'oauth:bitbucket';
+  options.authorizationURL = options.authorizationURL || 'https://bitbucket.org/site/oauth2/authorize';
+  options.tokenURL = options.tokenURL || 'https://bitbucket.org/site/oauth2/access_token';
+  options.customHeaders = options.customHeaders || {};
 
-  OAuthStrategy.call(this, options, verify);
+  //HACK: swap for OAuth2 property
+  if (options.consumerKey && !options.clientID) {
+    options.clientID = options.consumerKey;
+  }
+
+  //HACK: swap for OAuth2 property
+  if (options.consumerSecret && !options.clientSecret) {
+    options.clientSecret = options.consumerSecret;
+  }
+
+  if (!options.customHeaders['User-Agent']) {
+    options.customHeaders['User-Agent'] = options.userAgent || 'passport-bitbucket';
+    //HACK: requests need to fall back to Basic Auth (for access_token call)
```

Comment:
What's the motivation behind this? The fact that we do not currently support the use of query params for id:secret?